### PR TITLE
JENKINS-45887 : Fix TU and avoid workspace scan if not necessary

### PR DESCRIPTION
**Pull request by bguerin**
Fix #71

### CODE REVIEW

File: jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java [hunk 1]
```diff
@@ -71,12 +75,17 @@ public void process(@Nonnull final StepContext context, @Nonnull final Element m
         final Run run = context.get(Run.class);
         final Launcher launcher = context.get(Launcher.class);
 
-        final String pattern = "**/" + REPORTS_DIR + "/*";
-        final FilePath[] paths = workspace.list(pattern);
-        if (paths == null || paths.length == 0) {
+        boolean foundJGivenDependency = false;
+        List<MavenSpyLogProcessor.MavenDependency> dependencies = listDependencies(mavenSpyLogsElt, LOGGER);
+        for (MavenSpyLogProcessor.MavenDependency dependency : dependencies) {
+            if (dependency.artifactId.contains("jgiven")) {
```
Comment (cyrille-leclerc) [inline jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/publishers/JGivenTestsPublisher.java]: smart!

**Comment by cyrille-leclerc** [comment]
CI environment facing issues, let's move forward

**Comment by bguerin** [comment]
I ran a full mvn clean install on my laptop before pushing, it should be ok.

**Comment by cyrille-leclerc** [comment]
@bguerin same for me. It's an infra issue.

**Pull request merged**